boards/BMS/Src/Io/Io_CellVoltages.c

```diff
@@ -0,0 +1,156 @@
+#include "Io_CellVoltages.h"
+#include "Io_LTC6813.h"
+#include "Io_SharedSpi.h"
+#include "configs/App_CellConfigs.h"
+#include "configs/Io_LTC6813Configs.h"
+
+#define NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP 3U
+
+enum CellVoltageRegisterGroup
+{
+    CELL_VOLTAGE_REGISTER_GROUP_A,
+    CELL_VOLTAGE_REGISTER_GROUP_B,
+    CELL_VOLTAGE_REGISTER_GROUP_C,
+    CELL_VOLTAGE_REGISTER_GROUP_D,
+    CELL_VOLTAGE_REGISTER_GROUP_E,
+    CELL_VOLTAGE_REGISTER_GROUP_F,
+    NUM_OF_CELL_VOLTAGE_REGISTER_GROUPS
+};
+
+static const uint16_t cell_voltage_register_group_commands
+    [NUM_OF_CELL_VOLTAGE_REGISTER_GROUPS] = {
+        0x0400, // RDCVA
+        0x0600, // RDCVB
+        0x0800, // RDCVC
+        0x0A00, // RDCVD
+        0x0900, // RDCVE
+        0x0B00  // RDCVF
+    };
+
+static uint16_t cell_voltages[NUM_OF_CELL_MONITOR_ICS]
+                             [NUM_OF_CELLS_READ_PER_IC];
+
+/**
+ * Parse cell voltages received from the cell monitoring IC and perform PEC15
+ * checks.
+ * @param current_ic The current cell monitoring IC to parse cell voltages for.
+ * @param current_register_group The current register group on the given IC to
+ * parse cell voltages for.
+ * @param rx_cell_voltages The buffer containing the cell voltages read from the
+ * cell monitoring IC.
+ * @return EXIT_CODE_OK if the PEC15 check was successful. Else,
+ * EXIT_CODE_ERROR.
+ */
+static ExitCode Io_LTC6813_ParseCellsAndPerformPec15Check(
+    size_t                        current_ic,
+    enum CellVoltageRegisterGroup current_register_group,
+    uint8_t                       rx_cell_voltages[]);
+
+static ExitCode Io_LTC6813_ParseCellsAndPerformPec15Check(
+    size_t                        current_ic,
+    enum CellVoltageRegisterGroup current_register_group,
+    uint8_t                       rx_cell_voltages[])
+{
+    size_t cell_voltage_index = current_ic * NUM_OF_RX_BYTES;
+
+    for (size_t current_cell = 0U;
+         current_cell < NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP; current_cell++)
+    {
+        uint32_t cell_voltage =
+            (uint32_t)(rx_cell_voltages[cell_voltage_index]) |
+            (uint32_t)((rx_cell_voltages[cell_voltage_index + 1] << 8));
+
+        cell_voltages[current_ic]
+                     [current_cell +
+                      current_register_group *
+                          NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP] =
+                         (uint16_t)cell_voltage;
+
+        if (current_register_group == CELL_VOLTAGE_REGISTER_GROUP_F)
+        {
+            // Since 16 cells are monitored for each accumulator segment, ignore
+            // the last four bytes read back from the LTC6813 chip when reading
+            // from CELL_VOLTAGE_REGISTER_F. The cell voltage index is
+            // incremented by 6 to retrieve the PEC15 bytes for the register
+            // group.
+            cell_voltage_index += 6U;
+            break;
```

Comment:
"Since 16 cells are monitored for each accumulator segment and the number of cells monitored per each register group is 3, meaning only one cell is monitored by group f.......

Something like this maybe????

Reply:
I updated the comment here.